### PR TITLE
Update code to reflect previous refactoring

### PR DESCRIPTION
**Pull request by aanil**

### CODE REVIEW

File: dds_web/api/project.py [hunk 1]
```diff
@@ -392,7 +390,9 @@ def post(self):
                 else:
                     # If it is an existing user, add them to project.
                     add_user_result = AddUser.add_user_to_project(
-                        existing_user=existing_user, project=new_project.public_id, owner=owner
+                        existing_user=existing_user,
+                        project=new_project.public_id,
+                        owner=user.get("role") == "Project Owner",
```
Comment (inaod568) [inline dds_web/api/project.py:395]: Is the `add_user_to_project` code adapted to if the new added user has role Unit Admin or Personnel? I think there needs to be a few checks here before the user is added, and there needs to be a way of just adding the user to the unit if it's not a Researcher or Personnel.

Comment (aanil) [reply dds_web/api/project.py]: I've added a check for 'Project Owner' and 'Researcher' when adding users to a project. I think that adding users to a Unit needs to be a different function and not mixed in with the project addition functionality.

Comment (inaod568) [reply dds_web/api/project.py]: Great! Would be great if we could add some tests for when attempting to add Unit Personnel / Admins, to check that it fails.

Comment (aanil) [reply dds_web/api/project.py]: Added a few tests!

Comment (inaod568) [reply dds_web/api/project.py]: Nice! Thanks

File: tests/test_project_creation.py [hunk 2]
```diff
@@ -399,3 +411,17 @@ def test_create_project_with_invited_users(client):
     assert response.json and response.json.get("user_addition_statuses")
     for x in response.json.get("user_addition_statuses"):
         assert "Invitation sent" in x
+
+
+def test_create_project_with_unsuitabl_(client):
```
Comment (inaod568) [inline tests/test_project_creation.py:416]: A little typo

File: tests/test_user_add.py [hunk 3]
```diff
@@ -268,3 +268,18 @@ def test_existing_user_change_ownership_same_permissions(client):
...
+    response = client.post(
+        tests.DDSEndpoint.USER_ADD,
+        headers=tests.UserAuth(tests.USER_CREDENTIALS["unituser"]).post_headers(),
+        data=json.dumps(user_with_unsuitable_role),
+        content_type="application/json",
+    )
+    assert response.status == "403 FORBIDDEN"
+
+
+existing_research_user
```
Comment (inaod568) [inline tests/test_user_add.py:285]: Remove the last line?

File: tests/test_project_creation.py (hunk 2 above)
Comment (aanil) [reply tests/test_project_creation.py]: Oops!

File: tests/test_user_add.py (hunk 3 above)
Comment (aanil) [reply tests/test_user_add.py]: Probably a bit hasty in pushing the last commit 😅

Comment (aanil) [reply tests/test_user_add.py]: Changed it now!

**Review by inaod568 — APPROVED** [review]

**Comment by inaod568** [comment]
@aanil Looks great! Just waiting for the CodeQL to pass

**Pull request merged**